lib/views/product_detail.dart

```diff
@@ -0,0 +1,245 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:thirumathikart_app/controllers/product_detail_controller.dart';
+import 'package:thirumathikart_app/models/product.dart';
+import 'package:thirumathikart_app/config/themes.dart';
+import 'package:thirumathikart_app/views/products.dart';
+
+class ProductDetail extends GetView<ProductDetailsController> {
+  final Product product;
+  final List<Product> productList;
+  const ProductDetail(this.product, this.productList, {Key? key})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final productDetailsController = Get.put(ProductDetailsController());
```

Comment:
Do not initalize controller here . create your own bindings and do it. Refer other pages for it.